src/legacy/core_plugins/ui_metric/server/routes/api/ui_metric.ts

```diff
@@ -27,15 +26,27 @@ export async function storeReport(server: any, report: Report) {
   const { callWithInternalUser } = server.plugins.elasticsearch.getCluster('admin');
   const internalRepository = getSavedObjectsRepository(callWithInternalUser);
 
-  const metricKeys = Object.keys(report.uiStatsMetrics);
-  return Promise.all(
-    metricKeys.map(async key => {
-      const metric = report.uiStatsMetrics[key];
+  const uiStatsMetrics = report.uiStatsMetrics ? Object.entries(report.uiStatsMetrics) : [];
+  const userAgents = report.userAgent ? Object.entries(report.userAgent) : [];
+  return Promise.all([
+    ...userAgents.map(async ([key, metric]) => {
+      const { userAgent } = metric;
+      const savedObjectId = `${key}:${userAgent}`;
+      return await internalRepository.create(
+        'ui-metric',
+        { count: 1 },
+        {
+          id: savedObjectId,
+          overwrite: true,
+        }
+      );
+    }),
+    ...uiStatsMetrics.map(async ([key, metric]) => {
       const { appName, eventName } = metric;
       const savedObjectId = `${appName}:${eventName}`;
-      return internalRepository.incrementCounter('ui-metric', savedObjectId, 'count');
-    })
-  );
+      return await internalRepository.incrementCounter('ui-metric', savedObjectId, 'count');
+    }),
+  ]);
```

Comment:
nit picking, but I always find inline `map` into `Promise` hard to read, because people aren't used to `map` functions being `async` ( I don't think this is just me, I've seen others fall for this trap before).

The way I usually get around this is by using function which has a type signature that makes this clearer, such as: `Array<UserAgent> => Array<Promise<UserAgentSavedObject>>`, or something like that.

This is *definitely* a personal preference, and not a blocker.

Reply:
The preview here is wrong, it should have included this block in it: https://github.com/elastic/kibana/pull/50015/files/b768d190ec09be7323a980f7615f950e601f8376#diff-d4bc2abf31e5023c5205a2a7dec4bf3bR29

Github's multiline comment is flaky